### PR TITLE
Upgrade OpenVINO Toolkit v2024.4

### PR DESCRIPTION
**Pull request by jatinwadhwa921**
Upgrade OpenVINO Toolkit v2024.4

### CODE REVIEW

File: onnxruntime/core/providers/openvino/ov_versions/capability.cc [hunk 1]
```diff
@@ -35,16 +35,16 @@ GetCapability::GetCapability(const GraphViewer& graph_viewer_param,
     device_type_ = "CPU";
     if (enable_qdq_optimizer) npu_qdq_optimizer_enabled = true;
   }
-#if OPENVINO_VERSION_MAJOR == 2024 && OPENVINO_VERSION_MINOR == 0
-  data_ops_ = new DataOps(graph_viewer_, V_2024_0, device_type_, npu_qdq_optimizer_enabled);
-#elif OPENVINO_VERSION_MAJOR == 2024 && OPENVINO_VERSION_MINOR == 1
+#if OPENVINO_VERSION_MAJOR == 2024 && OPENVINO_VERSION_MINOR == 1
   data_ops_ = new DataOps(graph_viewer_, V_2024_1, device_type_, npu_qdq_optimizer_enabled);
```
Comment (preetha-intel) [inline onnxruntime/core/providers/openvino/ov_versions/capability.cc:39]: We can remove 2024.1 check as we only commit for latest three releases of OV

**Review by vthaniel — COMMENTED** [review]
CPP Unit tests on CPU are passing

File: onnxruntime/core/providers/openvino/ov_versions/capability.cc (hunk 1 above)
Comment (jatinwadhwa921) [reply onnxruntime/core/providers/openvino/ov_versions/capability.cc:39]: fixed

Comment (vthaniel) [reply onnxruntime/core/providers/openvino/ov_versions/capability.cc:39]: I think you have removed 2024.0 (which should have been removed in last release itself)
We need to remove 2024.1 also right?

**Pull request merged**